File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#
from typing import List
# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        for idx, val in enumerate(nums):
            tempTarget = target - val
            if tempTarget in nums and nums.index(tempTarget) != idx:
                return [idx, nums.index(tempTarget)]

# @lc code=end

"""
time: O(n)
space: O(n)
"""

File: 1.two-sum.java
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
  public int[] twoSum(int[] nums, int target) {
    int[] answer = new int[2];

    outer: for (int i = 0; i < nums.length - 1; i++) {
      for (int j = i + 1; j < nums.length; j++) {
        if (nums[i] + nums[j] == target) {
          answer[0] = i;
          answer[1] = j;
          break outer;
        }
      }
    }

    return answer;
  }
}
// @lc code=end

